src/sampling/regular.jl

```diff
@@ -172,4 +172,28 @@ end
 function sample(rng::AbstractRNG, grid::CartesianGrid,
                 method::RegularSampling)
   sample(rng, boundingbox(grid), method)
+end
+
+function sample(::AbstractRNG, torus::Torus{T},
+                method::RegularSampling) where {T}
```

Comment:
let's add this sampling method in a separate PR as suggested with proper tests and more careful review. This PR should only introduce the Torus type with the appropriate constructors and acessor functions.